src/test/groovy/org/osiam/resources/exceptions/OsiamExceptionHandlerSpec.groovy

```diff
@@ -26,89 +26,69 @@ package org.osiam.resources.exceptions
 
     OsiamExceptionHandler exceptionHandler = new OsiamExceptionHandler()
     def static IRRELEVANT = 'irrelevant'
-    WebRequest request = Mock()
 
-    def 'generated JsonErrorResult contains provided code and description'() {
-        when:
-        def errorResult = new OsiamExceptionHandler.JsonErrorResult('code', 'description')
-        then:
-        errorResult.error_code == 'code'
-        errorResult.description == 'description'
-    }
 
     def 'generating a response entity works'() {
         when:
-        def result = exceptionHandler.handleConflict(new NullPointerException(IRRELEVANT), request)
+        def result = exceptionHandler.defaultExceptionHandler(new NullPointerException(IRRELEVANT))
         then:
-        result.getStatusCode() == HttpStatus.CONFLICT
-        (result.getBody() as OsiamExceptionHandler.JsonErrorResult).error_code == HttpStatus.CONFLICT.name()
-        (result.getBody() as OsiamExceptionHandler.JsonErrorResult).description == IRRELEVANT
+        result.error_code == HttpStatus.CONFLICT.name()
+        result.description == "An unexpected error occurred"
     }
 
-    def 'status is set to ResourceNotFound when ResourceNotFoundException occurs'() {
+    def 'status is set to NOT_FOUND when ResourceNotFoundException occurs'() {
         when:
-        def result = exceptionHandler.handleConflict(new ResourceNotFoundException(IRRELEVANT), request)
+        def result = exceptionHandler.handleResourceNotFoundException(new ResourceNotFoundException(IRRELEVANT))
         then:
-        result.getStatusCode() == HttpStatus.NOT_FOUND
-        (result.getBody() as OsiamExceptionHandler.JsonErrorResult).error_code == HttpStatus.NOT_FOUND.name()
+        result.error_code == HttpStatus.NOT_FOUND.name()
+        result.description == IRRELEVANT
     }
 
     def "status is set to NOT_IMPLEMENTED when java.lang.UnsupportedOperationException occurs"() {
         when:
-        def result = exceptionHandler.handleConflict(new UnsupportedOperationException(IRRELEVANT), request)
+        def result = exceptionHandler.handleUnsupportedOperation(new UnsupportedOperationException(IRRELEVANT))
         then:
-        result.getStatusCode() == HttpStatus.NOT_IMPLEMENTED
-        (result.getBody() as OsiamExceptionHandler.JsonErrorResult).error_code == HttpStatus.NOT_IMPLEMENTED.name()
+        result.error_code == HttpStatus.NOT_IMPLEMENTED.name()
+        result.description == IRRELEVANT
     }
 
     def "status is set to I_AM_A_TEAPOT when org.osiam.resources.exceptions.SchemaUnknownException occurs"() {
         when:
-        def result = exceptionHandler.handleConflict(new SchemaUnknownException(), request)
+        def result = exceptionHandler.handleSchemaUnknown(new SchemaUnknownException())
         then:
-        result.getStatusCode() == HttpStatus.I_AM_A_TEAPOT
-        (result.getBody() as OsiamExceptionHandler.JsonErrorResult).error_code == HttpStatus.I_AM_A_TEAPOT.name()
```

Comment:
Ups, sorry. Didn't see it [moved to line 65](https://github.com/osiam/resource-server/pull/65/files#diff-1f816b83e22a3a93094a5ecb55c4ccd2R65)